Sources/Error Handling/PurchasesError.swift

```diff
@@ -39,7 +40,29 @@ extension PurchasesError {
     /// let errorCode = error as? ErrorCode
     /// ```
     var asPublicError: PublicError {
```

Comment:
My main concern with this approach is whether we only expose PublicError everywhere and/or if we use this method somewhere else... Need to research it a bit more.